tavern/internal/cryptocodec/cryptocodec.go

```diff
@@ -0,0 +1,195 @@
+package cryptocodec
+
+import (
+	"bytes"
+	"crypto/ecdh"
+	"crypto/rand"
+	"errors"
+	"log"
+	"runtime"
+	"strconv"
+	"sync"
+
+	"github.com/cloudflare/circl/dh/x25519"
+	"golang.org/x/crypto/chacha20poly1305"
+	"google.golang.org/grpc/encoding"
+)
+
+// TODO: Switch to a gomap and mutex.
+var session_pub_keys = sync.Map{}
+
+// TODO: Should we make this a random long byte array in case it gets used anywhere to avoid encrypting data with a week key? - Sliver handles errors in this way.
+var FAILURE_BYTES = []byte{}
```

Comment:
Should this be a random byte array so we can avoid a known static key in failure cases?